testbed/testbed/utils/graphsync.go

```diff
@@ -2,98 +2,129 @@ package utils
 
 import (
 	"context"
-	"fmt"
-	"time"
-
-	p2p "github.com/libp2p/go-libp2p-core"
-	"go.uber.org/fx"
 
+	"github.com/ipfs/go-blockservice"
 	"github.com/ipfs/go-cid"
 	"github.com/ipfs/go-graphsync"
 	gsimpl "github.com/ipfs/go-graphsync/impl"
 	"github.com/ipfs/go-graphsync/network"
 	"github.com/ipfs/go-graphsync/storeutil"
 	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	offline "github.com/ipfs/go-ipfs-exchange-offline"
 	files "github.com/ipfs/go-ipfs-files"
-	"github.com/ipfs/go-ipfs/core/node/helpers"
-	unixfsFile "github.com/ipfs/go-unixfs/file"
+	format "github.com/ipfs/go-ipld-format"
+	"github.com/ipfs/go-merkledag"
+	unixfile "github.com/ipfs/go-unixfs/file"
+	"github.com/ipfs/go-unixfs/importer/helpers"
+	"github.com/pkg/errors"
 
+	allselector "github.com/hannahhoward/all-selector"
 	"github.com/ipld/go-ipld-prime"
 	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
-	basicnode "github.com/ipld/go-ipld-prime/node/basic"
-	ipldselector "github.com/ipld/go-ipld-prime/traversal/selector"
-	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
+	"github.com/libp2p/go-libp2p-core/host"
 	"github.com/libp2p/go-libp2p-core/peer"
 )
 
-// Graphsync constructs a graphsync
-func Graphsync(lc fx.Lifecycle, mctx helpers.MetricsCtx, host p2p.Host, bs blockstore.GCBlockstore) graphsync.GraphExchange {
-	ctx := helpers.LifecycleCtx(mctx, lc)
+type GraphsyncNode struct {
```

Comment:
Quick question: The same way we have a testcase with Bitswap operating in isolation and within an IPFS node, would it make sense to have the same for Graphsync? The goal of the previous Graphsync testcase was to evaluate the use of Graphsync within an IPFS node as an injected dependency. 

Should we add an option in the IPFS node transfer to enable the use of Graphsync or Bitswap as exchange interfaces?

Reply:
I mean, the test of IPFS technically uses Bitswap, because that's all IPFS uses for most of its APIs. But really it's a test of the holistic IPFS stack fetching files. Eventually that should include GraphSync as well. So I wouldn't leave in a test of Graphsync w/ IPFS for now, cause that's not a real production integration of Graphsync into the overall IPFS API. But when we eventually ship Graphsync as part of IPFS, then the current IPFS test will naturally reflect that.